src/trainer_card.c

```diff
@@ -0,0 +1,1857 @@
+        StringExpandPlaceholders(sData->var_395, gText_Var1DarkGreyShadowLightGrey);
+    }
+}
+
+static void PrintPokeblockStringOnCard(void)
+{
+    if (sData->cardType != CARD_TYPE_FR && sData->trainerCard.pokeblocksWithFriends)
+        PrintString(3, gText_PokeblocksWithFriends, sData->var_395, gUnknown_0856FB0F);
+}
+
+static void PrintContestNumOnCard(void)
+{
+    if (sData->cardType != CARD_TYPE_FR && sData->trainerCard.contestsWithFriends)
+        ConvertIntToDecimalStringN(sData->var_3DB, sData->trainerCard.contestsWithFriends, 1, 5);
+}
+
+static void PrintContestStringOnCard(void)
+{
+    if (sData->cardType != CARD_TYPE_FR && sData->trainerCard.contestsWithFriends)
+        PrintString(4, gText_WonContestsWFriends, sData->var_3DB, gUnknown_0856FB0F);
+}
+
+static void PrintBattleFacilityNumsOnCard(void)
+{
+    switch (sData->cardType)
+    {
+    case CARD_TYPE_RUBY:
+        if (sData->hasBattleTowerWins)
+        {
+            ConvertIntToDecimalStringN(gStringVar1, sData->trainerCard.battleTowerWins, 1, 4);
+            ConvertIntToDecimalStringN(gStringVar2, sData->trainerCard.battleTowerStraightWins, 1, 4);
+            StringExpandPlaceholders(sData->var_421, gText_WSlashStraightSlash);
+        }
+        break;
+    case CARD_TYPE_EMERALD:
+        if (sData->trainerCard.frontierBP)
+        {
+            ConvertIntToDecimalStringN(gStringVar1, sData->trainerCard.frontierBP, 1, 5);
+            StringExpandPlaceholders(sData->var_421, gText_Var1DarkLightGreyBP);
+        }
+        break;
+    case CARD_TYPE_FR:
+        break;
+    }
+}
+
+static void PrintBattleFacilityStringOnCard(void)
+{
+    switch (sData->cardType)
+    {
+    case CARD_TYPE_RUBY:
+        if (sData->hasBattleTowerWins)
+            PrintString(5, gText_BattleTower, sData->var_421, gUnknown_0856FB0C);
+        break;
+    case CARD_TYPE_EMERALD:
+        if (sData->trainerCard.frontierBP)
+            PrintString(5, gText_BattlePtsWon, sData->var_421, gUnknown_0856FB0F);
+        break;
+    case CARD_TYPE_FR:
+        break;
+    }
+}
+
+static void sub_80C4140(void)
+{
+    u8 i;
+    u8 buffer[] = {0x05, 0x06, 0x07, 0x08, 0x09, 0x0a};
+    u8 buffer2[] = {0x00, 0x04, 0x08, 0x0c, 0x10, 0x14};
+
+    if (sData->cardType == CARD_TYPE_FR)
+    {
+        for (i = 0; i < 6; i++)
+        {
+            if (sData->trainerCard.monSpecies[i])
+            {
+                u8 monSpecies = sub_80D30A0(sData->trainerCard.monSpecies[i]);
+                WriteSequenceToBgTilemapBuffer(3, 16 * i + 224, buffer2[i] + 3, 15, 4, 4, buffer[monSpecies], 1);
+            }
+        }
+    }
+}
+
+static void sub_80C41D8(void)
+{
+    u8 i;
+
+    CpuSet(gMonIconPalettes, sData->var_468, 0x60);
+    switch (sData->trainerCard.var_4E)
+    {
+    case 0:
+        break;
+    case 1:
+        TintPalette_CustomTone(sData->var_468, 96, 0, 0, 0);
+        break;
+    case 2:
+        TintPalette_CustomTone(sData->var_468, 96, 500, 330, 310);
```

Comment:
These should be fixed point values

Reply:
All of them?

Reply:
Just the last three (the R, G, and B coefficients)